dcpy/utils/s3.py

```diff
@@ -38,15 +38,14 @@
 MAX_FILE_COUNT = 50
 
 
-@dataclass
-class Metadata:
+class Metadata(BaseModel):
```

Comment:
Why the change from dataclass to pydantic basemodel?

Reply:
I was trying to figure out how to go from Metadata to dictionary and for some reason things like `dataclass.asdict()`  wasn't working

so I changed it to a BaseModel to use`.model_dump()`. seems like we're using BaseModel in a lot of other places too

Reply:
Yeah I at times wonder if it'd just be cleaner to just use pydantic, even if it's overkill for some of our cases.

Reply:
muddying the water there is also a `dataclasses` submodule of pydantic